canonical/ubuntu-advantage-client
# Handle reverse dependencies

## Proposed Commit Message
Handle reverse dependencies

Properly sort service disable/enable order based either on required or dependent services necessities.

PS: This PR is also updating both required and dependent services to explicit define which `UAEntitlment` classes that rely upon.

## Test Steps
Run the modified unit tests and verify that the `ros` integration tests are still working as expected

## Desired commit type


## Checklist
